### PR TITLE
Few small fixes so tools=no and target=release compiles

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Made a few small fixes so Godot compiles again with these options (only tested on my mac)

### CODE REVIEW

File: scene/3d/baked_light_instance.cpp [hunk 1]
```diff
@@ -1695,12 +1695,16 @@ void BakedLight::create_debug_mesh(DebugMode p_mode) {
...
 	add_child(mmi);
+#ifdef TOOLS_ENABLED
 	if (get_tree()->get_edited_scene_root()==this){
 		mmi->set_owner(this);
 	} else {
 		mmi->set_owner(get_owner());
 
 	}
+#else
+    mmi->set_owner(get_owner());
```
Comment (akien-mga) [inline scene/3d/baked_light_instance.cpp]: Spaces... :)

File: scene/3d/gi_probe.cpp [hunk 2]
```diff
@@ -1314,12 +1314,16 @@ void GIProbe::_create_debug_mesh(Baker *p_baker) {
...
 	add_child(mmi);
+#ifdef TOOLS_ENABLED
 	if (get_tree()->get_edited_scene_root()==this){
 		mmi->set_owner(this);
 	} else {
 		mmi->set_owner(get_owner());
 
 	}
+#else
+    mmi->set_owner(get_owner());
```
Comment (akien-mga) [inline scene/3d/gi_probe.cpp]: Same here.

File: scene/3d/baked_light_instance.cpp [hunk 3]
```diff
@@ -1695,12 +1695,16 @@ void BakedLight::create_debug_mesh(DebugMode p_mode) {
 	MultiMeshInstance *mmi = memnew( MultiMeshInstance );
 	mmi->set_multimesh(mm);
 	add_child(mmi);
+#ifdef TOOLS_ENABLED
 	if (get_tree()->get_edited_scene_root()==this){
```
Comment (akien-mga) [inline scene/3d/baked_light_instance.cpp]: @reduz Shouldn't you just expose `SceneTree::get_edited_scene_root()` for non-tools builds and have it return null? Every single time you add such a test you forget the `#ifdef TOOLS_ENABLED`, so I guess in your mind it's supposed to work already and return false for the condition.

Comment (bojidar-bg) [inline scene/3d/baked_light_instance.cpp]: @akien-mga Or maybe we should do this?:
```
#ifndef TOOLS_ENABLED
#define get_tree()->get_edited_scene_root() NULL
#endif
```

Comment (akien-mga) [inline scene/3d/baked_light_instance.cpp]: @bojidar-bg Sounds a bit hackish to me :)

**Pull request merged**